include/effolkronium/random.hpp

```diff
@@ -642,6 +651,88 @@ namespace effolkronium {
             return dist( engine_instance( ) );
         }
 
+        /**
+        * \brief Return a random iterator from given map container by
+        *        utilizing the values of the map container as weights
+        *        for weighted random number generation
+        * \param Key The Key type for this version of 'get' method
+        *        Type should be '(THIS_TYPE)::common' struct
+        * \param map_container A container that has mapped_type,
+        *        value_type and key_type defined
+        * \note return the end iterator if the iterator is empty or total weight is equals to sum
+        */
+        template<
+            typename Key,
+            class MapContainer
+        >
+        static auto get(const MapContainer& map_container) -> typename std::enable_if<
+            details::is_map<MapContainer>::value &&
+            details::is_iterator<decltype(std::begin(map_container))>::value &&
+            (details::is_uniform_int<typename MapContainer::mapped_type>::value || details::is_byte<typename MapContainer::mapped_type>::value) &&
+            std::is_same<Key, common>::value,
```

Comment:
`common` type is reserved for common type randoms
I think it would be better to add another key type 'weight' for this feature

Reply:
Done!